### PR TITLE
Adding GPDB specific expression for `raw_expression_tree_walker`.

### PR DESCRIPTION
**Pull request by karthijrk**
This is to address the PR comment https://github.com/greenplum-db/gpdb/pull/2404#discussion_r115579525

Note: This PR is not intended to be merged into master but will first be merged into a cte_merge branch.

### CODE REVIEW

File: src/backend/optimizer/util/walkers.c [hunk 1]
```diff
@@ -1488,11 +1488,32 @@ raw_expression_tree_walker(Node *node, bool (*walker) (), void *context)
...
+		case T_OuterJoinInfo:
+		case T_DMLActionExpr:
+		case T_PartOidExpr:
+		case T_PartSelectedExpr:
+		case T_PartDefaultExpr:
+		case T_PartBoundExpr:
+		case T_PartBoundInclusionExpr:
+		case T_PartBoundOpenExpr:
+		case T_PartListRuleExpr:
+		case T_PartListNullTestExpr:
```
Comment (hsyuan) [inline src/backend/optimizer/util/walkers.c]: Will raw parse tree contain these node?

Comment (karthijrk) [inline src/backend/optimizer/util/walkers.c]: I don't know about that yet. We need to find out when we get there. We don't use raw_expr_tree_walker yet. By comparing with expr_tree_walker, these are the node type that does not exist in Postgres.

**Comment by karthijrk** [comment]
I made the changes based on how GPDB's expression_tree_walker has changed. Thanks Heikki for noticing that. This seems to be used by recursive CTE. When I looked at the postgres, walker callback handles only some specific expression and for others, it calls the `raw_expression_tree_walker`. So based on this, I think like other cases, I should call the walker callback itself.

**Comment by karthijrk** [comment]
@heikki I updated the changes based on your comments. Please let me know if it is fine now.

File: src/backend/optimizer/util/walkers.c (hunk 1 above)
Comment (hsyuan) [inline src/backend/optimizer/util/walkers.c]: I think `Part****Expr` these are not parse nodes. I think all the nodes can be found in `parsenodes.h`

**Comment by karthijrk** [comment]
Actually, I can find the nodes in Postgres' parse node that is not part of raw_expression_tree_walker. The comments in it are not helpful. So I am going to close this PR without merging it. In my opinion, we can address GPDB specific expression later when we get there.

**Pull request closed without merging**